### PR TITLE
Fixes issues with PHP 8.1 compatibility found by WebApi tests - part 2

### PR DESCRIPTION
**Pull request by karyna-t**
### Description (*)
This PR provides fixes for the errors found by WebApi Tests on PHP 8.1 build (E_ALL reporting level).
(part 2)

### Related Issues
1. magento/magento2#34653

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by karyna-tsymbal-atwix** [comment]
@magento run Static tests

**Comment by karyna-tsymbal-atwix** [comment]
@magento run Static tests

**Comment by karyna-tsymbal-atwix** [comment]
@magento run Static tests

**Comment by karyna-tsymbal-atwix** [comment]
@magento run Static Tests, Unit Tests, Semantic Version Checker

**Comment by karyna-tsymbal-atwix** [comment]
@magento run Static Tests, Unit Tests, Semantic Version Checker

**Comment by karyna-tsymbal-atwix** [comment]
@magento run Static Tests, Unit Tests, Semantic Version Checker

**Comment by karyna-tsymbal-atwix** [comment]
@magento run Static Tests, Semantic Version Checker

**Comment by xmav** [comment]
@magento run all tests

**Comment by xmav** [comment]
@karyna-tsymbal-atwix please check failed tests

**Comment by karyna-tsymbal-atwix** [comment]
Static test fail in `app/code/Magento/Sales/Model/Order.php` is fixed in the scope of another PR:
https://github.com/magento/magento2/pull/34703/files#diff-c8d66fa5975f0f5940ac131e828f9b7bd742603d7fcbe236b036a305525e4008L68

**Comment by karyna-tsymbal-atwix** [comment]
@magento run all tests

**Comment by karyna-tsymbal-atwix** [comment]
@magento run all tests

File: app/code/Magento/Sales/Test/Unit/Model/OrderTest.php [hunk 1]
```diff
@@ -403,9 +403,6 @@ public function testGetCustomerName(array $expectedData)
         $this->order->setCustomerFirstname($expectedData['first_name']);
         $this->order->setCustomerSuffix($expectedData['customer_suffix']);
         $this->order->setCustomerPrefix($expectedData['customer_prefix']);
-        $this->scopeConfigMock->expects($this->exactly($expectedData['invocation']))
```
Comment (xmav) [inline app/code/Magento/Sales/Test/Unit/Model/OrderTest.php:406]: Could you please add case to cover "middle" name too? It seems `getCustomerMiddlename()` may return `null` 
https://github.com/magento/magento2/blob/9f43821af62ab177e44b51b112ff8d32d9b82813/app/code/Magento/Sales/Model/Order.php#L2044

**Review by xmav — CHANGES_REQUESTED** [review]

File: app/code/Magento/Sales/Test/Unit/Model/OrderTest.php (hunk 1 above)
Comment (karyna-t) [reply app/code/Magento/Sales/Test/Unit/Model/OrderTest.php:406]: Hi @xmav , I've added changes to the test, please check.

please also check this change [setup/src/Magento/Setup/Model/BatchInsert.php](https://github.com/magento/magento2/pull/34689/files#diff-7c879d7e7310d11559221d4be3be7349d8f1933c8f30c315a87249d7623e586fR88)

**Comment by karyna-tsymbal-atwix** [comment]
@magento run all tests

**Comment by karyna-tsymbal-atwix** [comment]
@magento run all tests

**Review by xmav — APPROVED** [review]

**Comment by xmav** [comment]
pending delivery https://github.com/magento-commerce/magento2ce/pull/7291

**Pull request merged**